biggus/__init__.py

```diff
@@ -2125,8 +2187,20 @@ def _sliced_shape(shape, keys):
 
     """
     sliced_shape = []
-    # TODO: Watch out for more keys than shape entries.
-    for size, key in itertools.izip_longest(shape, keys):
+
+    # Figure out how each of the keys modifies the existing shape.
+    shape = list(shape)
+    keys = list(keys)
+    for key in keys:
+        # np.newaxis ... aka. None
+        if key is None:
+            sliced_shape.append(1)
+            continue
+
+        if not shape:
+            raise ValueError('too many keys')
```

Comment:
And [in](https://github.com/SciTools/biggus/blob/master/biggus/__init__.py#L537) [biggus](https://github.com/SciTools/biggus/blob/master/biggus/__init__.py#L698) [too](https://github.com/SciTools/biggus/blob/master/biggus/__init__.py#L737). :unamused: